secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  RecoveryPhraseStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/10/2022.
+//
+
+import Foundation
+
+/// Zcash implementation of the keychain that is not universal but designed to deliver functionality needed by the wallet itself.
+/// All the APIs should be thread safe according to official doc:
+/// https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
+/// For seamless usage the interface is designed to deal with simple types:
+/// `Seed Phrase: String` - possible to `store` and `retrive` from the keychain. `Delete` operation is also available, use it with caution.
+/// `Birthday: BlockHeight` - possible to `store` and `retrive` from the keychain. `Delete` operation is also available, use it with caution.
+final class RecoveryPhraseStorage {
+    static let `default`:RecoveryPhraseStorage = RecoveryPhraseStorage()
+    
+    enum KeychainError: Error, Equatable {
+        case duplicate
+        case unknown(OSStatus)
+        case dataEncodingFailed
+        case dataDecodingFailed
+        case noDataFound
+    }
+
+    private static let zcashSeedPhrase = "zcashSeedPhrase"
+    private static let zcashBirthday = "zcashBirthday"
+    
+    // MARK: - Seed Phrase Public Interface
+    
+    func setSeedPhrase(_ phrase: String) throws {
+        guard let data = phrase.data(using: .utf8) else {
+            throw KeychainError.dataEncodingFailed
+        }
+            
+        try setData(data, forKey: RecoveryPhraseStorage.zcashSeedPhrase)
+    }
+
+    func seedPhrase() throws -> String {
+        guard let data = data(forKey: RecoveryPhraseStorage.zcashSeedPhrase) else {
+            throw KeychainError.noDataFound
+        }
+        
+        return String(decoding: data, as: UTF8.self)
+    }
+    
+    /// Use carefully:  Deletes seed phrase from the keychain!!!
+    @discardableResult
+    func deleteSeedPhrase() -> Bool {
+        return deleteData(forKey: RecoveryPhraseStorage.zcashSeedPhrase)
```

Comment:
You don't need to use the return keyword here.

Reply:
done